ansible_collections/arista/avd/roles/eos_designs/doc/network-services-v4.0.md

```diff
@@ -0,0 +1,652 @@
+# Network Services Variables - VRFs/VLANs
+
+- The network services variables provide an abstracted model to create L2 and L3 network services across the fabric.
+- The network services are grouped by tenants. The definition of a tenant may vary between organizations.
+  - e.g. Tenants can be organizations or departments.
+- The tenant shares a common vni range for mac vrf assignment.
+- The filtering model allows for granular deployment of network service to the fabric leveraging the tenant name and tags applied to the service definition.
+  - This allows for the re-use of SVIs and VLANs across the fabric.
+
+## Variables and Options
+
+### Tenants Keys
+
+```yaml
+# Define network services keys, to define grouping of network services.
+# This provides the ability to define various keys of your choice to better organize/group your data.
+# This should be defined in top level group_var for the fabric.
+network_services_keys:
+  - name: < key_1 >
+  - name: < key_2 >
+```
+
+```yaml
+# Example
+# The below key/pair values are the role defaults.
+network_services_keys:
+  - name: tenants
+```
+
+```yaml
+# On mlag leafs, an SVI interface is defined per vrf, to establish iBGP peering. | Required (when mlag leafs in topology)
+# The SVI id will be derived from the base vlan defined: mlag_ibgp_peering_vrfs.base_vlan + (vrf_id or vrf_vni) - 1
+# Depending on the values of vrf_id / vrf_id it may be required to adjust the base_vlan to avoid overlaps or invalid vlan ids.
```

Comment:
Any reason to write ```vrf_id``` twice? I saw this to be vrf_vni in other doc.

Reply:
We support mpls networks now, so we need a neutral vrf_id. So we have vrf_id and vrf_vni as two distinct keys. I did not do any changes in that regard in this PR.

Reply:
I was referring to the line33. Just the documentation.

Reply:
Missed that sry. This was just copied from the regular network-services.md. I only updated the data models I have already converted. I guess this was fixed after I did the copy. Any way this doc will not be copied back again. It will be migrated key by key to the original when we are releasing the new data models.

Reply:
Yes, this was fixed in PR #1749 - this documentation issue is resolved.